File: pkg/handler/package.go
package handler

import (
	"net/http"

	"github.com/nektro/aquila/pkg/db"
	"github.com/nektro/aquila/pkg/handler/controls"
	etc "github.com/nektro/go.etc"
	"github.com/nektro/go.etc/htp"
)

func Package(w http.ResponseWriter, r *http.Request) {
	c := htp.GetController(r)
	user := controls.GetUserOptional(r)
	repo := controls.GetURemote(c, r)
	owner := controls.GetUUser(c, r, repo)
	pkg := controls.GetUPackage(c, r, owner)

	writePageResponse(w, r, "/package.hbs", map[string]interface{}{
		"aquila_version": etc.Version,
		"user":           user,
		"repo":           repo,
		"owner":          owner,
		"pkg":            pkg,
		"versions":       db.Version{}.AllByPackage(pkg),
	})
}

File: src/handler/package.zig
const std = @import("std");
const string = []const u8;
const http = @import("apple_pie");

const _internal = @import("./_internal.zig");

pub const Args = struct { remote: u64, user: string, package: string };

pub fn get(_: void, response: *http.Response, request: http.Request, captures: ?*const anyopaque) !void {
    const args = @ptrCast(*const Args, @alignCast(@alignOf(Args), captures));

    const alloc = request.arena;
    const u = try _internal.getUserOp(response, request);
    const r = try _internal.reqRemote(request, response, args.remote);
    const o = try _internal.reqUser(request, response, r, args.user);
    const p = try _internal.reqPackage(request, response, o, args.package);
    const v = try p.versions(alloc);

    // calling inline yields 'error: cannot store runtime value in compile time variable'
    const readme = try _internal.renderREADME(alloc, v[0]);

    try _internal.writePageResponse(alloc, response, request, "/version.pek", .{
        .aquila_version = @import("root").version,
        .page = "version",
        .title = try std.fmt.allocPrint(alloc, "{d}/{s}/{s}", .{ r.id, o.name, p.name }),
        .user = u,
        .repo = r,
        .owner = o,
        .package = p,
        .versions = v,
        .version = v[0],
        .readme = readme,
    });
}
